### PR TITLE
5908 fixes for release 3.2 (Send Request button)

### PR DESCRIPTION
**Pull request by SamirTouira**
issue: https://github.com/blockframes/blockframes/issues/5908#issuecomment-853749072

### CODE REVIEW

File: libs/landing/src/lib/shell/shell.component.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export class LandingFooterComponent { }
 export class LandingShellComponent {
   public submitted = false;
   public appName = getAppName(getCurrentApp(this.routerQuery));
+  public buttonText = 'Send Request'
```
Comment (Mussuranae) [inline libs/landing/src/lib/shell/shell.component.ts:44]: miss a `;`

File: libs/landing/src/lib/shell/shell.component.ts [hunk 2]
```diff
@@ -70,15 +71,22 @@ export class LandingShellComponent {
 
   /** Triggers when a user click on the button from LearnMoreComponent.  */
   public sendRequest(form: FormGroup) {
+
     if (form.invalid) {
       this.snackBar.open('Please fill the required informations.', 'close', { duration: 2000 });
+      this.buttonText = 'Send Request';
```
Comment (Mussuranae) [inline libs/landing/src/lib/shell/shell.component.ts:77]: Do you really need this one as you put a default text for the button ?

File: libs/landing/src/lib/shell/shell.component.html [hunk 3]
```diff
@@ -86,7 +86,7 @@ <h3>Schedule a demo with our team!</h3>
             <tel-input [form]="form.get('phoneNumber')" placeholder="Phone number"></tel-input>
           </mat-form-field>
         </article>
-        <button [disabled]="submitted" fxFlexAlign="center" mat-flat-button color="accent" type="submit">Send Request</button>
+        <button [disabled]="submitted" fxFlexAlign="center" mat-flat-button color="accent" type="submit">{{buttonText}}</button>
```
Comment (RemcoSimonides) [inline libs/landing/src/lib/shell/shell.component.html:89]: Add space `{{ buttonText }}`

File: libs/landing/src/lib/shell/shell.component.ts (hunk 1 above)
Comment (RemcoSimonides) [inline libs/landing/src/lib/shell/shell.component.ts:44]: `;` at the end of the line

File: libs/landing/src/lib/shell/shell.component.ts [hunk 4]
```diff
@@ -70,15 +71,22 @@ export class LandingShellComponent {
...
       return;
     }
+
     try {
+      if (form.status === "VALID"){
+        this.buttonText = 'Sending Request...';
+      }
```
Comment (RemcoSimonides) [inline libs/landing/src/lib/shell/shell.component.ts:82]: It happens that status is not valid here?
Line 75 you check for `form.invalid` already

Comment (Mussuranae) [reply libs/landing/src/lib/shell/shell.component.ts:82]: Agree with Remco, I'm not sure you need this check. And I think you can also write `if (form.valid)` (for a next time ;) )

File: libs/landing/src/lib/shell/shell.component.ts (hunk 2 above)
Comment (SamirTouira) [reply libs/landing/src/lib/shell/shell.component.ts:76]: Indeed, no need finally, thanks 👍

File: libs/landing/src/lib/shell/shell.component.ts (hunk 4 above)
Comment (SamirTouira) [reply libs/landing/src/lib/shell/shell.component.ts:82]: Alright, thank you!

**Comment by SamirTouira** [comment]
I changed that 👍

**Review by Mussuranae — APPROVED** [review]
good job !

**Pull request merged**